src/components/Check/CheckerControls.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useCallback, useState } from 'react'
+import {
+  TextInput,
+  useLayout,
+  GU,
+  Info,
+  // @ts-ignore
+} from '@aragon/ui'
+import { useHistory } from 'react-router-dom'
+import { fontWeight } from '../../style/font'
+import BrandButton from '../BrandButton/BrandButton'
+import ConverterFormControls from '../Migrate/Converter/ConverterFormControls'
+import styled from 'styled-components'
+import { Case, Switch } from 'react-if'
+import { optionsInfo } from '../../token-info/options'
+
+const FORTY_DIGITS_HEX = /^0x[0-9a-fA-F]{40}$/s
+type ComponentState = 'init' | 'error' | 'options' | 'no options'
+
+export function BaseCheckerFormControls(): JSX.Element {
+  const history = useHistory()
+  const { layoutName } = useLayout()
+  const [address, setAddress] = useState('')
+  const [state, setState] = useState<ComponentState>('init')
+  const [options, setOptions] = useState(0)
+
+  const handleNavigateHome = useCallback(() => {
+    history.push('/')
+  }, [history])
+
+  const stackedButtons = layoutName === 'small'
+
+  const handleAddressChange = useCallback((event) => {
+    const value = event.target.value
+    setAddress(value)
+  }, [])
+
+  const handleSubmit = () => {
+    const isValidAddress = FORTY_DIGITS_HEX.test(address)
+    if (!isValidAddress) return setState('error')
```

Comment:
use ether's function `isAddress` to check for valid address.
https://docs.ethers.io/v5/api/utils/address/#utils-isAddress

`isAddress()` treats zero address as valid, check to see if it should be invalid, and add that check yourself.